website/src/controllers/api-v3/user.js

```diff
@@ -41,6 +41,65 @@ api.getUser = {
   },
 };
 
+/**
+ * @api {get} /user/anonymized
+ * @apiVersion 3.0.0
+ * @apiName UserGetAnonymized
+ * @apiGroup User
+ * @apiSuccess {Object} object The object { user, tasks }
+ **/
+api.getUserAnonymized = {
+  method: 'GET',
+  middlewares: [authWithHeaders(), cron],
+  url: '/user/anonymized',
+  async handler (req, res) {
+    let user = res.locals.user.toJSON();
+    user.stats.toNextLevel = common.tnl(user.stats.lvl);
+    user.stats.maxHealth = common.maxHealth;
+    user.stats.maxMP = res.locals.user._statsComputed.maxMP;
+    delete user.apiToken;
+    if (user.auth) {
+      delete user.auth.local;
+      delete user.auth.facebook;
+    }
+    delete user.newMessages;
+    delete user.profile;
+    delete user.purchased.plan;
+    delete user.contributor;
+    delete user.invitations;
+    delete user.items.special.nyeReceived;
+    delete user.items.special.valentineReceived;
+    delete user.webhooks;
+    delete user.achievements.challenges;
+    _.forEach(user.inbox.messages, (msg) => {
+      msg.text = res.t('inboxMessageText');
+    });
+    _.forEach(user.tags, (tag) => {
+      tag.name = res.t('tag');
+      tag.challenge = res.t('challenge');
+    });
+    // HEREHERE
+    let query = { userId: user._id, $or: [{ type: 'todo', completed: false },
+                                           { type: { $in: ['habit', 'daily', 'reward'] } }],
+                };
+    let tasks = await Tasks.Task.find(query).exec();
+    function cleanChecklist (task) {
```

Comment:
I'd move this function outside of the router handler, it doesn't make sense to recreate it every time